### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by nataliacza**

### CODE REVIEW

File: BlogProject.Dtos/Accounts/UserRoleDto.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace BlogProject.Dtos.Accounts;
+
+public class UserRoleDto
```
Comment (Arebisc) [inline BlogProject.Dtos/Accounts/UserRoleDto.cs:3]: Can be removed

File: BlogProject.Services/Accounts/GenerateJwtToken.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+// TO DO
+// interface do jwt token generator - change name
+// add authClaims - add separate method
+// IOptions - configuration
+
+
+namespace BlogProject.Services.Accounts;
+
+public class GenerateJwtToken : ITokenGenerator, IUserClaims
```
Comment (Arebisc) [inline BlogProject.Services/Accounts/GenerateJwtToken.cs:19]: `JwtTokenGenerator`

File: BlogProject.Services/Accounts/GenerateJwtToken.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+{
+    private readonly JwtConfiguration _jwtSettings;
+
+    public GenerateJwtToken(
+        IOptions<JwtConfiguration> jwtSettings)
+    {
+        _jwtSettings = jwtSettings.Value;
+    }
+
+    public List<Claim> UserClaims(IdentityUser user)
```
Comment (Arebisc) [inline BlogProject.Services/Accounts/GenerateJwtToken.cs:29]: To private method `GetUserClaims`

File: BlogProject.Services/Accounts/GenerateJwtToken.cs (hunk 2 above)
Comment (Arebisc) [reply BlogProject.Services/Accounts/GenerateJwtToken.cs:19]: Interface `IUserClaims` can be moved inside `ITokenGenerator`

File: BlogProjectRefactor/Configuration/ConnectionStringsConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+﻿namespace BlogProject.Web.Configuration;
+
+public class ConnectionStringsConfiguration
```
Comment (Arebisc) [inline BlogProjectRefactor/Configuration/ConnectionStringsConfiguration.cs:3]: This is not used. Can be removed

File: BlogProjectRefactor/Controllers/PostsController.cs [hunk 5]
```diff
@@ -3,11 +3,11 @@
...
 using System.Linq;
-using System.Collections.ObjectModel;
-using System.Collections.Generic;
+using Microsoft.AspNetCore.Authorization;
 
 namespace BlogProject.Web.Controllers;
 
+
 [Route("api/posts")]
 public class PostsContoller : ControllerBase
```
Comment (Arebisc) [inline BlogProjectRefactor/Controllers/PostsController.cs:12]: Can be authorized here

File: BlogProject.Services/Accounts/GenerateJwtToken.cs (hunk 2 above)
Comment (nataliacza) [reply BlogProject.Services/Accounts/GenerateJwtToken.cs]: Token generator rebuild. Separate GetJwtClaims method created and used in GenerateJwtToken.

File: BlogProject.Services/Accounts/GenerateJwtToken.cs (hunk 3 above)
Comment (nataliacza) [reply BlogProject.Services/Accounts/GenerateJwtToken.cs]: Token generator rebuild.

File: BlogProjectRefactor/Configuration/ConnectionStringsConfiguration.cs (hunk 4 above)
Comment (nataliacza) [reply BlogProjectRefactor/Configuration/ConnectionStringsConfiguration.cs]: Removed.

File: BlogProjectRefactor/Controllers/PostsController.cs (hunk 5 above)
Comment (nataliacza) [reply BlogProjectRefactor/Controllers/PostsController.cs:13]: Done. Authorization applied to all endpoints.

File: BlogProject.Services/Accounts/GenerateJwtToken.cs (hunk 2 above)
Comment (nataliacza) [reply BlogProject.Services/Accounts/GenerateJwtToken.cs]: Class renamed to JwtTokenGenerator.

**Pull request merged**